userspace/falco/yaml_helper.h

```diff
@@ -26,17 +26,54 @@ limitations under the License.
 #endif
 #include <yaml-cpp/yaml.h>
 #include <string>
+#include <utility>
 #include <vector>
 #include <list>
 #include <set>
 #include <iostream>
 #include <fstream>
+#include <type_traits>
+#include <regex>
 
 #include "config_falco.h"
 
 #include "event_drops.h"
 #include "falco_outputs.h"
 
+class yaml_helper;
+
+class yaml_visitor {
```

Comment:
`yaml_visitor` is a small visitor API for yaml files, to call a callback function on each scalar value.
It is private to `yaml_helper` since it receives non-constant `YAML::Node`s (thus it can modify the parsed yaml structure).